src/lightning_trainable/trainable.py

```diff
@@ -22,6 +22,7 @@ class TrainableHParams(HParams):
     max_epochs: int | None
     max_steps: int = -1
     optimizer: str = "adam"
+    betas: tuple | list = (0.9, 0.999)
```

Comment:
I think it is good to be able to change this, but we should rather make optimizer a dict (optionally), and retrieve kwargs from that -- otherwise we end up adding parameters of all different kinds of optimizers to a huge single HParam

Reply:
Makes sense. There are already optimizer parameters in the HParam (`weight_decay`, `learning_rate`). I suggest to move weight decay to the new kwargs dict. Learning rate as well or keep it explicit?

Reply:
I don't mind `weight_decay` and `learning_rate` as all optimizers support them, but it is indeed cleaner to put them inside. But really, I don't care.

Reply:
I am in favour of moving both into the `optimizer` dict, it is more consistent and not overly verbose. Still, I think using just `optimizer="adam"` should be allowed. Another concern: What do we call the optimizer name in the dict? `name` or `kind` come to mind, but `algorithm` is more technically accurate.